app/components/datepicker/DatepickerDirective.js

```diff
@@ -7,185 +7,184 @@
-         */
-        function openEndDate() {
-            vm.endDate.opened = true;
-            vm.dateOptions.maxDate = vm.initialDateOptions.maxDate;
-            vm.dateOptions.minDate = vm.dts;
-        }
-
-        /**
-         * Will be fired after the start and the end date was chosen.
-         */
-        function onChangeDatepicker(){
-            vm.dateString = getFormattedDateString(vm.dts, vm.dte);
-            performDateSearch();
-        }
-
-        function getFormattedDateString(minDate, maxDate) {
-            return '[' + minDate.toISOString().replace('.000Z','') + ' TO ' +
-              maxDate.toISOString().replace('.000Z','') + ']';
-        }
-
-        function stringToDate(dateString) {
-            var dateArray = dateString.split(' TO ');
-            if (angular.isString(dateString) && dateArray.length === 2) {
-                dateArray[0] = new Date(dateArray[0].slice(1,11));
-                dateArray[1] = new Date(dateArray[1].slice(0,10));
-                if (dateArray[0] === 'Invalid Date' || dateArray[0] === 'Invalid Date') {
+    .directive('datePicker', ['$rootScope', 'HeatMapSourceGenerator', 'InfoService',
+        function($rootScope, HeatMapSourceGenerator, InfoService) {
+            return {
+                link: datePickerFilterLink,
+                templateUrl: 'components/datepicker/datepicker.tpl.html',
+                restrict: 'EA',
+                scope: {}
+            };
+
+            function datePickerFilterLink(scope) {
+
+                var vm = scope;
+
+                vm.initialDateOptions = {
+                    minDate: new Date('2013-03-01'),
+                    maxDate: new Date('2013-04-01')
+                };
+
+                vm.dateOptions = {
+                    minDate: HeatMapSourceGenerator.filterObj.getSearchObj().minDate,
+                    maxDate: HeatMapSourceGenerator.filterObj.getSearchObj().maxDate,
+                    startingDay: 1,
+                    showWeeks: false
+                };
+
+                vm.dateString = getFormattedDateString(vm.dateOptions.minDate,
+                                                        vm.dateOptions.maxDate);
+
+                vm.startDate = {
+                    opened: false
+                };
+
+                vm.endDate = {
+                    opened: false
+                };
+
+                vm.onChangeDatepicker = onChangeDatepicker;
+
+                vm.showDatepickerInfo = showDatepickerInfo;
+
+                vm.openEndDate = openEndDate;
+
+                vm.openStartDate = openStartDate;
+
+                vm.onSubmitDateText = onSubmitDateText;
+
+                vm.slider = defaultSliderValue();
+
+                /**
+                 * Set initial values for min and max dates in both of datepicker.
+                 */
+                vm.setInitialDates = function(){
+                    vm.dts = vm.dateOptions.minDate;
```

Comment:
can you rename `dts` with `dateStart` or something similar. it is very confusing. we should have better names and not just abbreviations.